### PR TITLE
docs: page.waitFor options object defination

### PR DESCRIPTION
**Pull request by theeko**
Added options descriptiptions for page.waitFor.

### CODE REVIEW

**Comment by theeko** [comment]
signed

File: docs/api.md [hunk 1]
```diff
@@ -1895,6 +1895,12 @@ This is a shortcut for [page.mainFrame().url()](#frameurl)
 #### page.waitFor(selectorOrFunctionOrTimeout[, options[, ...args]])
 - `selectorOrFunctionOrTimeout` <[string]|[number]|[function]> A [selector], predicate or timeout to wait for
 - `options` <[Object]> Optional waiting parameters
+  - `visible` <[boolean]> wait for element to be present in DOM and to be visible, i.e. to not have `display: none` or `visibility: hidden` CSS properties. Defaults to `false`.
```
Comment (mathiasbynens) [inline docs/api.md]: What about the `hidden` attribute?

Maybe it's better to not explicitly list the examples (like `display: none` or `visibility: hidden`) here.

Comment (theeko) [reply docs/api.md]: Changed as requested. Examples are not listed anymore. Just shows default value and description.

File: docs/api.md [hunk 2]
```diff
@@ -1895,6 +1895,12 @@ This is a shortcut for [page.mainFrame().url()](#frameurl)
 #### page.waitFor(selectorOrFunctionOrTimeout[, options[, ...args]])
 - `selectorOrFunctionOrTimeout` <[string]|[number]|[function]> A [selector], predicate or timeout to wait for
 - `options` <[Object]> Optional waiting parameters
+  - `visible` <[boolean]> wait for element to be present in DOM and to be visible. Defaults to `false`.
+  - `timeout` <[number]> maximum time to wait for in milliseconds. Defaults to `30000` (30 seconds). Pass `0` to disable timeout. The default value can be changed by using the [page.setDefaultTimeout(timeout)](#pagesetdefaulttimeouttimeout) method.
+  - `hidden` <[boolean]> wait for element to not be found in the DOM or to be hidden. Defaults to `false`.
+  - `polling` <[string]|[number]> An interval at which the `pageFunction` is executed, defaults to `raf`. If `polling` is a number, then it is treated as an interval in milliseconds at which the function would be executed. If `polling` is a string, then it can be one of the following values:
+    - `raf` - to constantly execute `pageFunction` in `requestAnimationFrame` callback. This is the tightest polling mode which is suitable to observe styling changes.
```
Comment (mathiasbynens) [inline docs/api.md:1902]: missing type

File: docs/api.md [hunk 3]
```diff
@@ -1895,6 +1895,12 @@ This is a shortcut for [page.mainFrame().url()](#frameurl)
...
+    - `mutation` - to execute `pageFunction` on every DOM mutation.
```
Comment (mathiasbynens) [inline docs/api.md:1903]: missing type

Comment (theeko) [reply docs/api.md:1903]: Type for mutation and raf? This was how it is in other parts of docs. (frame.waitForFunction, page.waitForFunction etc)

**Review by mathiasbynens — APPROVED** [review]

**Comment by mathiasbynens** [comment]
LGTM but before we merge this, we should make the linter happy and make sure the CI checks pass.

**Comment by mathiasbynens** [comment]
> LGTM but before we merge this, we should make the linter happy and make sure the CI checks pass.

To do this, https://github.com/GoogleChrome/puppeteer/blob/1248a19135fe54b8fe42e15435c11c3c8aa949d2/lib/Page.js#L1111 needs to be updated accordingly.

**Comment by theeko** [comment]
Checks are 9/10.
Dont know what last error is (Travis CI - Pull Request Action required after 10m — Build Errored)
Can you guide me.

**Comment by mathiasbynens** [comment]
@theeko The remaining failure seems unrelated to your change. I've just restarted the build.

**Comment by mathiasbynens** [comment]
Thanks!

**Comment by theeko** [comment]
Thanks for guidance Mathias.

**Pull request merged**